bzl/toolchains/BUILD

```diff
@@ -0,0 +1,247 @@
+        "@rules_cc//cc/toolchains/features/legacy:build_interface_libraries",
+        "@rules_cc//cc/toolchains/features/legacy:dynamic_library_linker_tool",
+        "@rules_cc//cc/toolchains/features/legacy:strip_debug_symbols",
+        "@rules_cc//cc/toolchains/features/legacy:linkstamps",
+        "@rules_cc//cc/toolchains/features/legacy:output_execpath_flags",
+        "@rules_cc//cc/toolchains/features/legacy:runtime_library_search_directories",
+        "@rules_cc//cc/toolchains/features/legacy:library_search_directories",
+        "@rules_cc//cc/toolchains/features/legacy:libraries_to_link",
+        "@rules_cc//cc/toolchains/features/legacy:force_pic_flags",
+        "@rules_cc//cc/toolchains/features/legacy:user_link_flags",
+        "@rules_cc//cc/toolchains/features/legacy:legacy_link_flags",
+        "@rules_cc//cc/toolchains/features/legacy:linker_param_file",
+        "@rules_cc//cc/toolchains/features/legacy:fission_support",
+        "@rules_cc//cc/toolchains/features/legacy:sysroot",
+    ],
+)
+
+HOSTS = (
+    ("linux", "x86_64"),
+    # TODO
+    # ("win", "x86_64"),
+    # ("mac", "x86_64"),
+    # ("mac", "aarch64"),
+)
+
+_HOST_OS_CONSTRAINTS = {
+    "linux": "@platforms//os:linux",
+    "win": "@platforms//os:windows",
+    "mac": "@platforms//os:macos",
+}
+
+_HOST_CPU_CONSTRAINTS = {
+    "x86_64": "@platforms//cpu:x86_64",
+    "aarch64": "@platforms//cpu:aarch64",
+}
+
+[cc_toolchain(
+    name = "esp_clang{}-{}_toolchain_xtensa".format(host_os, host_cpu),
+    action_type_configs = [
+        "@esp_clang_{}-{}//:xtensa-esp32s3-elf-ar".format(host_os, host_cpu),
+        "@esp_clang_{}-{}//:clang".format(host_os, host_cpu),
+        "@esp_clang_{}-{}//:clang++".format(host_os, host_cpu),
+        "@esp_clang_{}-{}//:lld".format(host_os, host_cpu),
+        "@esp_clang_{}-{}//:xtensa-esp32s3-elf-objcopy".format(host_os, host_cpu),
+        "@esp_clang_{}-{}//:xtensa-esp32s3-elf-strip".format(host_os, host_cpu),
+    ],
+    args = select({
+        # "//bzl/platforms:xtensa": [
+        #     ":xtensa",
+        # ],
+        "//conditions:default": [],
+    }) + [
+        ":bazel_no_absolute_paths",
+        ":nostdlibxx",
+        ":nostartfiles",
+        ":sys_includes",
+    ],
+    compiler = "esp-clang",  # Useful for distinguishing gcc vs clang.
+    cxx_builtin_include_directories = [
+        # This just doesn't work. I don't know why.
+        # "%sysroot%/xtensa-esp32s3-elf/include/c++/11.2.0",
+        # "%sysroot%/xtensa-esp32s3-elf/include/c++/11.2.0/xtensa-esp32s3-elf",
+        # "%sysroot%/xtensa-esp32s3-elf/include/c++/11.2.0/backward",
+        # "%sysroot%/xtensa-esp32s3-elf/include",
+    ],
+    exec_compatible_with = [
+        _HOST_CPU_CONSTRAINTS[host_cpu],
+        _HOST_OS_CONSTRAINTS[host_os],
+    ],
+    # This is also equally useless, but needs to be specified or the compiler
+    # looks at /usr/include.
+    sysroot = "external/esp_clang_{}-{}".format(host_os, host_cpu),
```

Comment:
Nothing here works, new `rules_cc` doesn't any of these arguments, and the version is pinned to a pre-release the pico-SDK was built on. Big sad.